### DISCUSSION TITLE
Everything in routes.swift?

### DISCUSSION THREAD

**Discussion by jaysonng** [netreconlab/parse-server-swift#38] [Q&A] [answered]
Hi,

Currently all cloudcode needs to be in routes.swift? 

There's no way to separate code into their own files for maintainability?

thanks,
Jayson

**Comment by cbaker6** [comment, marked as answer]
This question is related to how to use Vapor as oppose to ParseServerSwift. Here’s a couple articles that shows how to register Vapor routes created in different files.

- https://alexandrecools.medium.com/vapor-routes-groups-and-collections-5ff920720317
- https://medium.com/@cweinberger/routing-in-vapor-4-b49ad909e33c

**Reply by jaysonng** [reply]
I actually saw those articles and reading through them. I was just curious because of how cloudcode / webhooks  is setup that maybe we can't have anything a level deeper from the host. ie host/xxx/score.

thanks! some more stuff to read on.

**Reply by cbaker6** [reply]
It shouldn’t matter how deep you make the route, `host/score/save/before` looks like:

https://github.com/netreconlab/parse-server-swift/blob/9619d3c4d8a1dd22aa034385d10a758becfe1851/Sources/ParseServerSwift/routes.swift#L92-L94

**Reply by jaysonng** [reply]
aaaaah that's right! thanks thanks :)

**Reply by jaysonng** [reply]
There may be a bug with routing and collections.

```
    func boot(routes: RoutesBuilder) throws {
        
        let lokalityGroup = routes.grouped("lokality")
        lokalityGroup.post("first", name:"first", use: getLokalityFirst)
    }
```

With this code, I get a webhook that has a destination address /first instead of the expected http://127.0.0.1/lokality/first

![Screenshot 2023-06-23 at 2 01 47 AM](https://github.com/netreconlab/parse-server-swift/assets/440445/892d9eec-8f7a-4192-a495-1868e719f8ea)

Bug right?

**Reply by cbaker6** [reply]
This doesn't look like a bug to me, `ParseServerSwift` doesn't know how to read "grouped" and doesn't extend `HTTPRoutesGroup`, it extends `RoutesBuilder` instead. You will have to define the path manually like I showed https://github.com/netreconlab/parse-server-swift/discussions/38#discussioncomment-6241393.

It's possible that using `grouped` is a feature request that can help build paths, if so, you would need to open an issue and ask for the feature. Then you can attempt to open a PR to add the feature and I can review.

**Reply by cbaker6** [reply]
Without a PR, you can probably get away with some variation of:

```swift
func boot(routes: RoutesBuilder) throws {
        
        let lokalityGroup = routes.grouped("lokality")
        let path = lokalityGroup.append("first")
        lokalityGroup.post(path, name:"first", use: getLokalityFirst)
 }
```

**Reply by jaysonng** [reply]
> It's possible that using grouped is a feature request that can help build paths, if so, you would need to open an issue and ask for the feature. Then you can attempt to open a PR to add the feature and I can review.

Right. Sorry. I should have said feature request. 

Ill do a Feature request but ill also try the append trick. 

Thank you,

**Reply by cbaker6** [reply]
@jaysonng it turns out you were correct and this is actually a bug in ParseServerSwift, will be fixed in #41
